src/rgw/rgw_metadata.h

```diff
@@ -379,7 +379,8 @@ class RGWMetadataManager {
   int lock_exclusive(string& metadata_key, timespan duration, string& owner_id);
   int unlock(string& metadata_key, string& owner_id);
 
-  int get_log_shard_id(const string& section, const string& key, int *shard_id);
+  int get_log_shard_id(const string& section, const string& key, int *shard_id,
+                       unsigned int max_shards = 0);
```

Comment:
does this last argument need to have a default `= 0`? i would prefer to have all callers pass a max_shards so we don't need to bother with `rgw_md_log_max_shards` at all here

Reply:
@cbodley Thanks. That function is called only in one place, so it is more reasonable to force the caller to pass a max_shards parameter. I will fix that.